platform/linux-generic/m4/odp_crypto.m4

```diff
@@ -0,0 +1,38 @@
+# ODP_CRYPTO
+# ----------
+# Select default crypto implementation
+AC_ARG_WITH([crypto],
+            [AS_HELP_STRING([--with-crypto],
+                            [Choose crypto implementation (openssl/armv8crypto/null)]
+                            [[default=with] (linux-generic)])],
+            [], [with_crypto=openssl])
+
+# Default to OpenSSL implementation if crypto is enabled
+AS_IF([test "x$with_crypto" = "xyes"], [with_crypto=openssl])
+
+# Default to Null implementation if crypto is disabled
+AS_IF([test "x$with_crypto" = "xno"], [with_crypto=null])
+AS_IF([test "${with_openssl}" = "no"], [with_crypto=null])
```

Comment:
I had to add this condition back because the x86 and Arm64 CI jobs for `--without-openssl --wtihout-pcap` were failing as it was defaulting to OpenSSL and wasn't handled. But this change looks much cleaner; I will modify it. Thanks Matias!